persistence/provider/memory/eventstore.go

```diff
@@ -125,18 +163,26 @@ func (cs *eventStoreChangeSet) stageSave(
 	}
 
 	cs.items = append(cs.items, item)
-
 	return next
 }
 
 // eventStoreDatabase contains data that is committed to the event store.
 type eventStoreDatabase struct {
-	items []*eventstore.Item
+	items   []*eventstore.Item
+	offsets map[string]uint64
 }
 
 // apply updates the database to include the changes in cs.
 func (db *eventStoreDatabase) apply(cs *eventStoreChangeSet) {
 	db.items = append(db.items, cs.items...)
+
+	if db.offsets == nil {
+		db.offsets = make(map[string]uint64)
+	}
+
+	for _, item := range cs.items {
+		db.offsets[item.Envelope.MetaData.MessageId] = item.Offset
```

Comment:
```suggestion
		db.offsets[item.ID()] = item.Offset
```

I put these `ID()` convenience methods on the items.  I should probs add one to `Envelope` and `Parcel` as well.